pysmt/optimization/optimizer.py

```diff
@@ -124,15 +124,15 @@ def _binary_bound(self, lb, ub):
             u = lb + abs(lb) + 1
         return (l + u + 1) // 2
 
-    def _bound(self, lb, ub, strategy):
+    def _bound(self, lb, ub, strategy, goal):
         if strategy == 'binary':
-            return self._binary_bound(lb, ub)
+            return self._binary_bound(lb, ub) #ok
         if strategy == 'ub':
```

Comment:
After this refactoring we shall call this strategy `linear` instead of `ub` (because maximization works by improving lower bounds, not upper bounds).